### PR TITLE
K1J-54: Implemented ready for sign end point

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/event/model/CertificateEventType.java [hunk 1]
```diff
@@ -15,7 +15,9 @@ public enum CertificateEventType {
   COMPLEMENT("complement-certificate", Constants.ACTION_TYPE_CREATION, null),
   ANSWER_COMPLEMENT("answer-complement-message", Constants.ACTION_TYPE_CREATION,
       "message-sent"),
-  REVOKED("revoked-certificate", Constants.ACTION_TYPE_CHANGE, "certificate-revoked");
+  REVOKED("revoked-certificate", Constants.ACTION_TYPE_CHANGE, "certificate-revoked"),
+  READY_FOR_SIGN("ready-for-sign-certificate", Constants.ACTION_TYPE_CHANGE,
+      "certificate-ready-for-sign");
```
Comment (mhornfeldt) [inline certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/event/model/CertificateEventType.java:20]: Need to check this

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/certificate/model/Certificate.java [hunk 2]
```diff
@@ -278,6 +279,20 @@ public void revoke(ActionEvaluation actionEvaluation, RevokedInformation revoked
         .build();
   }
 
+  public void readyForSign(ActionEvaluation actionEvaluation) {
+    if (this.status != Status.DRAFT) {
+      throw new IllegalStateException(
+          "Incorrect status '%s' - required status is '%s' to send".formatted(this.status,
```
Comment (ChrilleSigerhed) [inline certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/certificate/model/Certificate.java:285]: Update the text to something like "...to set ready for sign"

**Review by ChrilleSigerhed — CHANGES_REQUESTED** [review]
This looks great, well done!

The code reads well and follows a clear structure!

Only found a little typo error, everything else looked splendid to me!

Please also add readyForSign as a field on ```CertificateMetadataDTO``` in ```CertificateMetadataConverter```

```java
.readyForSign(certificate.readyForSign().readyForSignAt())
```

**Comment by mhornfeldt** [comment]
> This looks great, well done!
> 
> The code reads well and follows a clear structure!
> 
> Only found a little typo error, everything else looked splendid to me!
> 
> Please also add readyForSign as a field on `CertificateMetadataDTO` in `CertificateMetadataConverter`
> 
> ```java
> .readyForSign(certificate.readyForSign().readyForSignAt())
> ```

Found the missing conversion while checking the flow locally and added it but with an added null check, otherwise it won't work. Everything else seems to work locally.

**Review by ChrilleSigerhed — APPROVED** [review]
Looks great, well done!

**Pull request merged**